src/App.jsx

```diff
@@ -1,21 +1,31 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import AddTodoForm from "./AddTodoForm.jsx";
 import TodoList from "./TodoList.jsx";
 import './App.css';
 
+const useSemiPersistentState = () => {
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')) || []);
```

Comment:
Nice usage of an empty array here! Small solutions to unexpected behavior like what you encountered are an important part of creating Production-ready software.